src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1664,5 +1664,242 @@ internal static Exception CreateInvalidNameCharException(string name, int index,
+
+            destination[0] = 'f';
+            destination[1] = 'a';
+            destination[2] = 'l';
+            destination[3] = 's';
+            destination[4] = 'e';
+            charsWritten = 5;
+            return true;
+        }
+
+        internal static bool TryFormat(char value, Span<char> destination, out int charsWritten)
+        {
+            charsWritten = -1;
+            if (destination.Length < 1) return false;
+
+            destination[0] = value;
+            charsWritten = 1;
+            return true;
+        }
+
+        internal static bool TryFormat(decimal value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, NumberFormatInfo.InvariantInfo);
+        }
+
+        internal static bool TryFormat(sbyte value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out  charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(short value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(int value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(long value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(byte value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ushort value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(uint value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ulong value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(float value, Span<char> destination, out int charsWritten)
+        {
+            if (float.IsNegativeInfinity(value))
+            {
+                if (destination.Length < 4)
+                {
+                    charsWritten = -1;
+                    return false;
+                }
+                destination[0] = '-';
```

Comment:
I cannot understand this comment. We need to write "-INF" in case of IsNegativeInfinity

Reply:
ditto = same as before (that was about using "foo"`u8` syntax and copy for clarity)

Reply:
The point is - I understand it might be slightly faster but I don't believe it's worth making code couple of times longer just to safe few nano seconds

Reply:
@krwq I this better ?
```
internal static bool TryFormat(float value, Span<char> destination, out int charsWritten)
{
    ReadOnlySpan<char> valueSpan = default;

    if (float.IsNegativeInfinity(value))
    {
        valueSpan = "-INF";
    }
    else if (float.IsPositiveInfinity(value))
    {
        valueSpan = "INF";
    }
    else if (IsNegativeZero((double)value))
    {
        valueSpan = "-0";
    }

    if (valueSpan.IsEmpty)
        return value.TryFormat(destination, out charsWritten, "R", NumberFormatInfo.InvariantInfo);

    charsWritten = valueSpan.Length;
    return valueSpan.TryCopyTo(destination);
}
```